test/parallel/xyk-pallet.API.fees.bothTokens.test.ts

```diff
@@ -0,0 +1,340 @@
+      expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+
+  await testUser1.refreshAmounts(AssetWallet.AFTER);
+  const deductedMGATkns = testUser1
+      .getAsset(MGA_ASSET_ID)
+      ?.amountBefore.free.sub(
+        testUser1.getAsset(MGA_ASSET_ID)?.amountAfter.free!
+      );
+    const fee = cost.partialFee;
+    expect(deductedMGATkns).bnLte(fee);
+    expect(deductedMGATkns).bnGt(new BN(0));
+
+});
+
+test("xyk-pallet - Check required fee - User with KSM only", async () => {
+
+  //add pool's tokens for user.
+  await UserMintCurrencyToken(testUser1, sudo, firstCurrency, defaultCurrecyValue);
+
+  //add KSM tokens.
+  await testUser1.addKSMTokens(sudo);
+
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+
+  await (await getMangataInstance())
+    .buyAsset(
+      testUser1.keyRingPair,
+      firstCurrency.toString(),
+      secondCurrency.toString(),
+      new BN(100),
+      new BN(1000000)
+    )
+    .then((result) => {
+      const eventResponse = getEventResultFromMangataTx(result);
+      expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+
+  await testUser1.refreshAmounts(AssetWallet.AFTER);
+
+  const deductedKSMTkns = testUser1
+    .getAsset(KSM_ASSET_ID)
+    ?.amountBefore.free.sub(
+      testUser1.getAsset(KSM_ASSET_ID)?.amountAfter.free!
+    );
+
+  expect(deductedKSMTkns).bnGt(new BN(0));
+
+});
+
+test("xyk-pallet - Check required fee - User with very few MGA and some KSM", async () => {
+
+  //add pool's tokens for user.
+  await UserMintCurrencyToken(testUser1, sudo, firstCurrency, defaultCurrecyValue);
+
+   //add few MGX tokens.
+   await sudo.mint(
+    MGA_ASSET_ID,
+    testUser1,
+    new BN(100000)
+  );
+
+  //add some KSM tokens.
+  await testUser1.addKSMTokens(sudo);
+
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+
+  await (await getMangataInstance())
+    .buyAsset(
+      testUser1.keyRingPair,
+      firstCurrency.toString(),
+      secondCurrency.toString(),
+      new BN(100),
+      new BN(1000000)
+    )
+    .then((result) => {
+      const eventResponse = getEventResultFromMangataTx(result);
+      expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+
+  await testUser1.refreshAmounts(AssetWallet.AFTER);
+
+  const deductedMGATkns = testUser1
+    .getAsset(MGA_ASSET_ID)
+    ?.amountBefore.free.sub(
+      testUser1.getAsset(MGA_ASSET_ID)?.amountAfter.free!
+    );
+  const deductedKSMTkns = testUser1
+    .getAsset(KSM_ASSET_ID)
+    ?.amountBefore.free.sub(
+      testUser1.getAsset(KSM_ASSET_ID)?.amountAfter.free!
+    );
+   
+  expect(deductedMGATkns).bnEqual(new BN(0));
+  expect(deductedKSMTkns).bnGt(new BN(0));
+  
+});
+
+test("xyk-pallet - Check required fee - User with some MGA and very few KSM", async () => {
+
+  //add pool's tokens for user.
+  await UserMintCurrencyToken(testUser1, sudo, firstCurrency, defaultCurrecyValue);
```

Comment:
this line is duplicated across all the tests, its a good candidate to move it to the beforeEach